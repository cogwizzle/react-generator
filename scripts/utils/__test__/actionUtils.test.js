const {
  addAction,
  modifyAction,
  generateBaseFiles,
} = require('../actionUtils')

it('addAction', () => {
  const results = addAction('.', 'Hello World')
  expect(results).toEqual({
    type: 'add',
    path: '.',
    template: 'Hello World',
    skipIfExists: false,
  })
})

it('addAction with skip', () => {
  const results = addAction('.', 'Hello World', true)
  expect(results).toEqual({
    type: 'add',
    path: '.',
    template: 'Hello World',
    skipIfExists: true,
  })
})

it('modifyAction', () => {
  const results = modifyAction('.', /\n/, ';')
  expect(results).toEqual({
    type: 'modify',
    path: '.',
    pattern: /\n/,
    template: ';',
  })
})

it('generateBaseFiles', () => {
  const results = generateBaseFiles('.', 'js', 'css', 'component', false)
  expect(results).toMatchSnapshot()
})

it('generateBaseFiles with class component and semicolons', () => {
  const results = generateBaseFiles('.', 'js', 'css', 'class component', true)
  expect(results).toMatchSnapshot()
})

