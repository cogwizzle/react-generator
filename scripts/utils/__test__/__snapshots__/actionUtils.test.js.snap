// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateBaseFiles 1`] = `
Array [
  Object {
    "path": "./{{snakeCase name}}/{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "./{{snakeCase name}}/{{snakeCase name}}.css",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "./{{snakeCase name}}/{{snakeCase name}}.js",
    "pattern": /;\\\\n/g,
    "template": "
",
    "type": "modify",
  },
]
`;

exports[`generateBaseFiles with class component and semicolons 1`] = `
Array [
  Object {
    "path": "./{{snakeCase name}}/{{snakeCase name}}.js",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
  Object {
    "path": "./{{snakeCase name}}/{{snakeCase name}}.css",
    "skipIfExists": false,
    "template": undefined,
    "type": "add",
  },
]
`;
